code/modules/mob/living/carbon/slime/slime.dm

```diff
@@ -894,6 +894,62 @@ mob/living/carbon/slime/var/temperature_resistance = T0C+75
 	icon_state = "bottle14"
 	w_class = W_CLASS_TINY
 
+/obj/item/weapon/slimechange
+	name = "slime dye"
+	desc = "A bottle of dye that can be used on a slime to change it's color."
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "bottle14"
+	w_class = W_CLASS_TINY
+
+/obj/item/weapon/slimechange/attack(var/mob/living/M, mob/user as mob)
+	if(istype(M, /mob/living/carbon/slime))
+		var/mob/living/carbon/slime/OS = M
+		if(OS.stat)
+			to_chat(user, "<span class='warning'>You're unable to use the [src] on the dead slime!</span>")
+			return..()
+		var/new_color = input(user, "Choose a new color for the dye:", "Color Select") as anything in list("adamantine", "black", "blue", "bluespace", "cerulean", "darkpurple", "darkblue", "darkpurple", "gold", "grey", "lightpink", "metal", "oil", "orange", "pink", "purple", "pyrite", "red", "sepia", "silver", "yellow")
+		if(new_color)
+			var/adult = FALSE //Passinging on if it was an adult or not
+			if(istype(OS, /mob/living/carbon/slime/adult))
+				adult = TRUE
+			var/new_slime
+			if(new_color == "grey")
+				new_slime = /mob/living/carbon/slime
+			else
+				new_slime = text2path("/mob/living/carbon/slime[adult ? "/adult" : ""]/[new_color]")
+			var/mob/living/carbon/slime/NS = new new_slime(OS.loc)
+			//Passing on several other traits to make it almost exactly the same as the old one.
```

Comment:
You'll want to mind.transfer_to the new slime if the original slime has a mind

Reply:
resolved on line 949